### ISSUE TITLE
[TODO] Test IFC alignment file

### ISSUE THREAD

**Issue by pjanck** [tumcms/Open-Infra-Platform#568]
[File](https://github.com/buildingSMART/IFC4.x-IF/tree/main/IFC-files/Trimble) to test. Read in, post a screen shot. If there are errors, report and discuss them with product owner.

**Comment by Elvira2227** [comment]
I tried to test [TrimbleTestFile_Ver2.zip](https://github.com/buildingSMART/IFC4.x-IF/blob/main/IFC-files/Trimble/TrimbleTestFile_Ver2.zip)

First of all, I corrected Ifc Schema Name 'IFC4X3_RC4' (I use IFC4X3_RC4)

And I detected folowing errors:
![image](https://user-images.githubusercontent.com/76201952/194836525-21770639-1cc4-4917-981e-21257c98e189.png)

![image](https://user-images.githubusercontent.com/76201952/194835425-6646f75d-ec3b-4f9c-92ba-9296f6267906.png)
And similar errors but in text format
```
error in #413068=IFCRELCONTAINEDINSPATIALSTRUCTURE('18xrRHNsLBEABYlQI9bj7e',#6,$,$,(#378,#393,#398,#380,#387,#388,#374,#375,#381,#383,#382,#384,#385,#386,#394,#395,#396,#397,#399,#400),#338697);: Could not find reference with ID=338697
error in #413076=IFCRELCONTAINEDINSPATIALSTRUCTURE('1N5FIFlkP0rx67c0Qklhp8',#6,$,$,(#391,#390),#412832);: Could not find reference with ID=412832
error in #413077=IFCRELCONTAINEDINSPATIALSTRUCTURE('1152BtFF97ZAGiPXg8Iy2i',#6,$,$,(#392,#389),#412833);: Could not find reference with ID=412833
error in #413078=IFCRELAGGREGATES('2l$IDV_8z0cRzxMJnnwHIz',#6,$,$,#412834,(#412832,#412833,#338697));: Could not find reference with ID=412834
error in #413079=IFCRELCONTAINEDINSPATIALSTRUCTURE('0cyL60ct9FefI1HVsVlM5j',#6,$,$,(#412828,#345824,#346729,#411923,#347634,#349357,#350246,#351135,#352024,#353731,#354732,#355733,#356734,#411520,#410635,#409686,#408336,#407928,#408737,#408329,#406094,#404737,#405145,#407043,#404336,#404744,#403451,#402534,#400808,#401617,#401209,#401216,#397681,#398089,#397688,#397280,#399006,#395446,#394497,#394089,#394096,#396395,#393688,#390969,#390568,#399923,#383785,#383384,#382976,#384734,#383377,#391886,#390160,#374971,#369137,#360795,#358961,#362489,#361680,#381174,#382091,#375856,#377614,#376665,#363406,#362088,#364323,#376264,#372672,#387377,#388326,#374022,#372665,#376257,#371379,#378563,#386568,#386976,#386969,#379849,#380257,#379448,#379856,#370462,#369545,#368736,#369144,#373073,#385683,#362081,#358560,#359878,#372264,#365208,#358553,#389275,#390561,#392803,#365609,#366934,#366017,#365616,#367851,#412831,#338934),#412834);: Could not find reference with ID=412834
```
Unfortunately, compilation take a long time on my machine and I cannot get the end result.

**Comment by pjanck** [comment]
Thank you for the report. Please also try and test the `_Ver3`. These errors may have been improved upon between the two versions.

**Comment by Elvira2227** [comment]
Testing [TrimbleTestFile_Ver3.zip](https://github.com/buildingSMART/IFC4.x-IF/blob/main/IFC-files/Trimble/TrimbleTestFile_Ver3.zip) gives the same result.

Ifc Schema corrected to `IFC4X3_RC4`

Screen shot: 
![image](https://user-images.githubusercontent.com/76201952/194837791-74a41bef-d9dc-492e-80f8-15660c0738f2.png)
Next errors in text format:
```
error in #370632=IFCRELCONTAINEDINSPATIALSTRUCTURE('3Tq_B5xovC5euyxyrrskRS',#6,$,$,(#1766,#2022,#2107,#1801,#1920,#1937,#1696,#1713,#1818,#1852,#1835,#1869,#1886,#1903,#2039,#2056,#2073,#2090,#2124,#2141),#377);: Could not find reference with ID=377
error in #370633=IFCRELCONTAINEDINSPATIALSTRUCTURE('2vyOUYUhb3Sh4auDbHF5sr',#6,$,$,(#1988,#1971),#378);: Could not find reference with ID=378
error in #370634=IFCRELCONTAINEDINSPATIALSTRUCTURE('0aDGRaxHz20vpWcnZxIR5h',#6,$,$,(#2005,#1954),#379);: Could not find reference with ID=379
error in #370639=IFCRELCONNECTSPORTS('1qahumSzXCfwurCTq0Osye',#6,$,$,#1472,(#304),$);: invalid stoull argument
error in #370640=IFCRELAGGREGATES('2ygoW9icv1uPaU0LmbBNko',#6,$,$,#1473,(#378,#379,#377));: Could not find reference with ID=1473
error in #370641=IFCRELCONTAINEDINSPATIALSTRUCTURE('3_f1gILUvAgQG8eLEiPZt7',#6,$,$,(#341312,#341313,#341431,#304,#1472,#27512,#21165,#21126,#21206,#21624,#22544,#22076,#23012,#23480,#24442,#23961,#24923,#25140,#25037,#25733,#26919,#28105,#28153,#28201,#28297,#28249,#28345,#28441,#28393,#26326,#28489,#341401,#341416,#341773,#343083,#343065,#343101,#342776),#1473);: Could not find reference with ID=1473
```
And compilation is also to long and I cannot get any result.

**Comment by pjanck** [comment]
Right, thank you! This is interesting, especially, since some of the 'missing elements' are present in the file:

![grafik](https://user-images.githubusercontent.com/59165496/194839027-d886a825-5167-445c-8110-5c84adff6a4f.png)

I suspect that the currently committed schema (according to which you've generated the early binding library) is out-of-date. A separate issue incoming!

**Comment by pjanck** [comment]
After #569 is fixed, regenerate the early bind library, compile anew, grab a coffee, and try again. Do not forget to change the schema used in the file.
